core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/SearchResultGenerator.kt

```diff
@@ -32,23 +32,30 @@ class ZimSearchResultGenerator @Inject constructor(
   private val sharedPreferenceUtil: SharedPreferenceUtil,
   private val zimReaderContainer: ZimReaderContainer
 ) : SearchResultGenerator {
+  private var searchTerm = ""
```

Comment:
remove this, introducing mutable state to the class breaks functional purity

Reply:
Removed. Replaced with arguments